test/models/diary_entry_test.rb

```diff
@@ -25,26 +18,32 @@ def test_diary_entry_validations
   end
 
   def test_diary_entry_visible
-    assert_equal 5, DiaryEntry.visible.count
+    visible = create(:diary_entry)
+    hidden = create(:diary_entry, :visible => false)
+    assert_includes DiaryEntry.visible, visible
     assert_raise ActiveRecord::RecordNotFound do
-      DiaryEntry.visible.find(diary_entries(:deleted_entry).id)
+      DiaryEntry.visible.find(hidden.id)
     end
   end
 
   def test_diary_entry_comments
-    assert_equal 0, diary_entries(:normal_user_entry_1).comments.count
-    assert_equal 4, diary_entries(:normal_user_geo_entry).comments.count
+    diary = create(:diary_entry)
+    create(:diary_comment, :diary_entry => diary)
```

Comment:
Then again maybe we should leave it as is but add an assertion that there are zero comments before that create...